### PR TITLE
sqlmodel(dm): support generating UPDATE CASE WHEN

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/3408

### What is changed and how it works?

as title

#### Tests 

 - Unit test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by lance6716** [comment]
ptal @buchuitoudegou

File: dm/syncer/dml.go [hunk 1]
```diff
@@ -355,6 +355,8 @@ func genSQLMultipleRows(op sqlmodel.DMLType, dmls []*sqlmodel.RowChange) (querie
 	switch op {
 	case sqlmodel.DMLInsert, sqlmodel.DMLReplace, sqlmodel.DMLInsertOnDuplicateUpdate:
 		return sqlmodel.GenInsertSQL(op, dmls...)
+	case sqlmodel.DMLUpdate:
+		return sqlmodel.GenUpdateSQL(dmls...)
```
Comment (buchuitoudegou) [inline dm/syncer/dml.go:359]: > // Compared to GenInsertSQL with DMLInsertOnDuplicateUpdate, this function is
// slower and more complex, we should only use it when PK/UK is updated.

I noticed this comment in `GenUpdateSQL`, which indicates we should only use it when updating the primary keys, but it seems we didn't consider it in this PR. Would there be any problems?

Comment (lance6716) [reply dm/syncer/dml.go:359]: https://github.com/pingcap/tiflow/blob/5140371b5990c91845f79d16f0b2988785083586/dm/syncer/dml.go#L464-L471

this is already implemented

**Review by buchuitoudegou — APPROVED** [review]

**Comment by lance6716** [comment]
ptal @GMHDBJD @dsdashun

**Comment by lance6716** [comment]
ptal @GMHDBJD @dsdashun

File: dm/syncer/dml.go [hunk 2]
```diff
@@ -419,9 +420,24 @@ func genDMLsWithSameTable(op sqlmodel.DMLType, jobs []*job) ([]string, [][]inter
...
+			}
+			if lastTable != j.dml.TargetTableID() {
+				query, arg := genDMLsWithSameCols(op, groupDMLs)
+				queries = append(queries, query...)
+				args = append(args, arg...)
+
+				groupDMLs = groupDMLs[0:0]
+				lastTable = j.dml.TargetTableID()
+			}
+			groupDMLs = append(groupDMLs, j.dml)
```
Comment (GMHDBJD) [inline dm/syncer/dml.go:435]: Can we combine these lines with L447?

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Comment by GMHDBJD** [comment]
Have not measured the syntax, not sure if the performance has improved and there are no other side effects

File: dm/syncer/dml.go (hunk 2 above)
Comment (lance6716) [reply dm/syncer/dml.go:435]: Will refine the code in future. In this sprint plan to let CDC use this library with @asddongmen

**Pull request merged**